machine/v1/types_powervsprovider.go

```diff
@@ -0,0 +1,154 @@
+	// +optional
+	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
+
+	// More detail about Power VS service instance
+	// https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-creating-power-virtual-server
+
+	// serviceInstance is the reference to the Power VS service on which the server instance(VM) will be created.
+	// Power VS service is a container for all Power VS instances at a specific geographic region.
+	// serviceInstance can be created via IBM Cloud catalog or CLI.
+	// serviceInstance identifier (name or id) can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// +kubebuilder:validation:=Required
+	ServiceInstance PowerVSResource `json:"serviceInstance"`
+
+	// image is to identify the rhcos image uploaded to IBM COS bucket which is used to create the instance.
+	// image identifier (name or id) can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// +kubebuilder:validation:=Required
+	Image PowerVSResource `json:"image"`
+
+	// network is the reference to the Network to use for this instance.
+	// network identifier (name or id) can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// +kubebuilder:validation:=Required
+	Network PowerVSResource `json:"network"`
+
+	// keyPairName is the name of the KeyPair to use for SSH
+	// +kubebuilder:validation:=Required
+	KeyPairName string `json:"keyPairName"`
+
+	// systemType is the System type used to host the instance.
+	// systemType determines the number of cores and memory that is available.
+	// Few of the supported SystemTypes are s922,e880,e980
+	// e880 systemType available only in Dallas Datacenters
+	// e980 systemType available in Datacenters except Dallas and Washington
+	// Defaults to s922 which is generally available.
+	// +optional
+	SystemType string `json:"systemType,omitempty"`
+
+	// processorType is the VM instance processor type.
+	// It must be set to one of the following values: dedicated, capped or shared.
+	// dedicated: resources are allocated for a specific client, The hypervisor makes a 1:1 binding of a partition’s processor to a physical processor core.
+	// shared: shared among other clients
+	// capped: shared, but resources do not expand beyond those that are requested, the amount of CPU time is capped to the value specified for the entitlement.
+	// if the processorType is selected as dedicated, then processors value cannot be fractional
+	// Defaults to shared.
+	// +kubebuilder:validation:Enum:="dedicated";"shared";"capped"
+	// +optional
+	ProcessorType string `json:"processorType,omitempty"`
+
+	// processors is the number of virtual processors in a virtual machine.
+	// when the processorType is selected as dedicated the processors value cannot be fractional
+	// The minimum processors is 0.5
+	// Defaults to 0.5.
+	// +optional
+	Processors string `json:"processors,omitempty"`
+
+	// memoryGiB is the size of a virtual machine's memory, in GB.
```

Comment:
This should be GibiBytes right?
```suggestion
	// memoryGiB is the size of a virtual machine's memory, in GiB.
```